pycalphad/io/database.py

```diff
@@ -33,6 +33,34 @@ def _to_tuple(lst):
     return tuple(_to_tuple(i) if isinstance(i, list) else i for i in lst)
 
 
+class Specie(object):
+    """
+    A specie in the database.
+
+    Attributes
+    ----------
+    name : string
+        Name of the specie
+    constituents : dict
+        Dictionary of {element: quantity} where the element is a string and the quantity a float.
+    charge : int
+        Integer charge. Can be positive or negative.
+    """
+    def __init__(self, name, constituents, charge):
+        self.name = name
+        self.constituents = constituents
+        self.charge = charge
+
+    def __eq__(self, other):
+        """Two species are the same if their names and constituents are the same."""
+        if isinstance(other, self.__class__):
+            return (self.name == other.name) and (self.constituents == other.constituents)
+        else:
+            return False
+
+    def __hash__(self):
+        return hash(self.name)
```

Comment:
This is not a good implementation of hash because it will break putting species into sets and frozensets when some species have the same name but different charges. Or is the name an unambiguous descriptor?

Reply:
The TDB format differentiates species (for example Al+2 and Al+3) by their names, which are arbitrary but unique and unambiguous. In that case, the names are conventionally `AL+3` and `AL+2` (but again, are arbitrary). 

If I am understanding your meaning correctly, the tests should demonstrate this case for the species 
```
SPECIES O-2                         O1/-2!
SPECIES O1                          O!
```

which have the same chemical stoichometry but are differentiated by name.

Reply:
That answers my question, so it should be fine.